operator/webhooks/pod_mutating_webhook.go

```diff
@@ -144,8 +146,10 @@ func (a *PodMutatingWebhook) handleWorkload(ctx context.Context, logger logr.Log
 		return nil
 	}
 
-	workload.Spec.Version = pod.Annotations[common.VersionAnnotation]
-	workload.Spec.ResourceReference = a.getResourceReference(pod)
+	if !reflect.DeepEqual(workload.Spec, newWorkload.Spec) {
+		workload.Spec = newWorkload.Spec
+	}
+	
 	workload.Annotations[common.VersionAnnotation] = pod.Annotations[common.VersionAnnotation]
```

Comment:
don't we want to update all annotations and not just the version one? Or at least update pre- and post-deployment check annotations